sdk/core/core/test/cmocka/test_span.c

```diff
@@ -11,6 +11,61 @@
 
 #include <_az_cfg.h>
 
+void az_span_from_string_non_ascii()
+{
+  az_span span = AZ_SPAN_FROM_STR("12汉字345");
+  assert_int_equal(11, az_span_length(span));
+}
+
+void az_span_from_string_non_ascii_latin()
+{
+  az_span span = AZ_SPAN_FROM_STR("è");
+  assert_int_equal(2, az_span_length(span));
+}
+
+void az_span_from_string_non_ascii_roundtrip()
+{
+  az_span span = AZ_SPAN_FROM_STR("12汉字345");
+  char label2[20] = "abcdefghijklmnopq";
+  char* labelPtr = label2;
+
+  assert_int_equal(AZ_OK, az_span_to_str(labelPtr, 20, span));
```

Comment:
think you can do `assert_return_code(az_span_to_str(...), AZ_OK)`  ,  but it is the same, so, however you want it is fine

Reply:
Also, you don't really  need to create labelPtr on line 30.
Just call `az_span_to_str(label2, 20, span)` and it should work